analysis/pseudobulk-bulk-prediction/exploratory-notebooks/residuals-annotated-genes.Rmd

```diff
@@ -0,0 +1,190 @@
+---
+title: "Are we confounded by genes that aren't real?"
+author: "Stephanie J. Spielman"
+date: "`r Sys.Date()`"
+output:
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    code_folding: hide
+---
+
+This notebook compares model residual distributions between gene with symbol annotations and genes without. 
+We have wondered if genes without annotation, which are less likely to be "real" genes, are biasing our downstream analyses.
+
+
+## Setup
+
+
+```{r setup}
+renv::load()
+
+library(ggplot2)
+theme_set(theme_bw())
+```
+
+### Paths
+
+```{r paths}
+model_dir <- here::here("analysis", "pseudobulk-bulk-prediction", "results", "models")
+ensembl_symbol_df <- readr::read_tsv(
+  here::here("analysis", "bulk-deconvolution", "data", "reference", "ensembl_symbol.tsv"), 
+  show_col_types = FALSE
+)
+```
+
+### Functions
+
+This chunk defines a helper function for loading and preparing residual data.
+
+```{r}
+read_prepare_data <- function(file_pattern, ensembl_symbol_df) {
+  
+  model_files <- list.files(
+    path = model_dir,
+    full.names = TRUE,
+    pattern = file_pattern
+  ) |>
+    purrr::set_names(
+      \(f) {stringr::str_split_i(basename(f), pattern = "_", i = 1)}
+    )
+  
+  model_files |>
+    purrr::map(
+      \(model_file) {
+        model_data <- readr::read_rds(model_file)
+        
+        residuals_df <- model_data$data |> 
+          dplyr::mutate(residuals = unname(resid(model_data$model))) |>
+          dplyr::left_join(ensembl_symbol_df, by = "ensembl_id") |>
+          # If the join resulted in an NA, then it's not annotated
+          dplyr::mutate(status = ifelse(
+            is.na(gene_symbol), 
+            "not_annotated", 
+            "annotated"
+          ))
+    }) |>
+    purrr::list_rbind(names_to = "project_id")
+}
+```
+
+This chunk defines a helper function for plotting boxplots.
+```{r}
+make_boxplot <- function(df, yvar, title) {
+  ggplot(df) +
+    aes(x = sample_id, fill = status, y = {{yvar}}) + 
```

Comment:
I don't think I care too much about the per-sample distributions. I would rather see the overall distribution per project in more detail than a boxplot (e.g., histogram).